variables.tf

```diff
@@ -29,6 +29,36 @@ variable "bucket_logging" {
   default     = false
 }
 
+variable "block_public_access" {
+  description = "Block various forms of public access on a per bucket level"
+  type        = bool
+  default     = false
+}
+
+variable "block_public_access_acl" {
+  description = "Related to block_public_access. PUT Bucket acl and PUT Object acl calls will fail if the specified ACL allows public access. PUT Object calls will fail if the request includes an object ACL."
+  type        = bool
+  default     = false
+}
+
+variable "block_public_access_ignore_acl" {
+  description = "Related to block_public_access. Ignore public ACLs on this bucket and any objects that it contains."
+  type        = bool
+  default     = false
+}
+
+variable "block_public_access_policy" {
+  description = "Related to block_public_access. Reject calls to PUT Bucket policy if the specified bucket policy allows public access."
+  type        = bool
+  default     = false
+}
+
+variable "block_public_access_restrict_bucket" {
+  description = "Related to block_public_access. Only the bucket owner and AWS Services can access this buckets if it has a public policy."
+  type        = bool
+  default     = false
+}
+
```

Comment:
Not sure about the default values.  For now I don't think we should enable the `block_public_access` by default, as it might cause unexpected interruptions, but for the others I think we would want to enable them by default, and require them to be explicitly disabled if thats desired.  

As is I fear we will create a resource that will not actually block public access.